### DISCUSSION TITLE
Is it possible to create public GitHub packages by default for OSS projects?

### DISCUSSION THREAD

**Discussion by ethan42** [community/community#26700] [Packages]
<p>Hello!</p>
<p>I am part of a team creating content for OSS developers and are trying to provide them with GitHub action workflows that “just work”. However, we have stumbled on a roadblock. Imagine that we have a simple GitHub action as follows:</p>
<pre><code class="lang-auto">env:
  REGISTRY: ghcr.io
  IMAGE_NAME: ${{ github.repository }}

jobs:
  build:
    name: '${{ matrix.os }} shared=${{ matrix.shared }} ${{ matrix.build_type }}'
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest]
        shared: [false]
        build_type: [Release]
        include:
          - os: ubuntu-latest
            triplet: x64-linux

    steps:
      - uses: actions/checkout@v2

      - name: Log in to the Container registry
        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
        with:
          registry: ${{ env.REGISTRY }}
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}

      - name: Extract metadata (tags, labels) for Docker
        id: meta
        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
        with:
          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}

      - name: Build and push Docker image
        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
        with:
          context: .
          push: true
          tags: ${{ steps.meta.outputs.tags }}
          labels: ${{ steps.meta.outputs.labels }}
</code></pre>
<p>Nothing special above (and we love that!), you build your image (say <code>ethan42/oss</code>) and then you proceed to push it with your GITHUB_TOKEN to <a href="http://ghcr.io" rel="noopener nofollow ugc">ghcr.io</a> (<code>ghcr.io/ethan42/oss</code>) and copy-pasting this action to your github project “just works”, correctly pushing your image under your packages. However, now image that this is an open source project and we would like the image we create to be public. This is technically possible through <a href="https://docs.github.com/en/enterprise-cloud@latest/packages/learn-github-packages/configuring-a-packages-access-control-and-visibility#configuring-visibility-of-container-images-for-your-personal-account" rel="noopener nofollow ugc">multiple steps that are well hidden</a> within the documentation, but it definitely takes us away from the “copy-paste your action and it just works” mentality. We have been unable to find an option to tweak the default visibility for such packages or even update it through APIs during the GitHub action.</p>
<p>So, is anyone aware of a way to create public images within <a href="http://ghcr.io" rel="noopener nofollow ugc">ghcr.io</a> by default? Or perhaps a project setting that enables you to do that?</p>
<p>Thank you for the time!</p>

**Comment by queenofcorgis** [comment]
<p>Welcome to the community, <a class="mention" href="/ethan42">@ethan42</a> :tada: I’ve recategorized your post to our Actions subcategory so it will get better visibility among those able to assist :slight_smile:</p>

**Comment by ethan42** [comment, marked as answer]
<p>I was able to find a solution through GitHub’s excellent support channel, so posting here in case someone else is interested:</p>
<p>When a container package is published using the built-in GitHub Actions workflow token ( <code>${{ secrets.GITHUB_TOKEN }}</code> ), the published package should inherit the visibility of the associated repo.</p>
<p>For orgs, the ability to publish public container packages must be explicitly enabled using this setting.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/5/6/5657bf6f5bcc63cfa0bf145e5adee4b7f54ff03f.png" data-download-href="/uploads/short-url/cjP3NoILZKv4186mO4M7PXhPpHp.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1320532/181099293-9125bb2d-0217-4d60-973f-d052558163e4.png" alt="image" data-base62-sha1="cjP3NoILZKv4186mO4M7PXhPpHp" width="690" height="261" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/5/6/5657bf6f5bcc63cfa0bf145e5adee4b7f54ff03f_2_690x261.png, https://aws1.discourse-cdn.com/github/original/3X/5/6/5657bf6f5bcc63cfa0bf145e5adee4b7f54ff03f.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/5/6/5657bf6f5bcc63cfa0bf145e5adee4b7f54ff03f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/6/5657bf6f5bcc63cfa0bf145e5adee4b7f54ff03f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">992×376 32.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>Users should <em>always</em> be able to publish public container packages, but wasn’t working until a few days ago. You should find this is now fixed.</p>
<p>If you run your workflow from a user owned public repo, you should find the published image is automatically public. For org owned public repos, you will need to make sure the <code>Public</code> option is checked.</p>
